src/lib/pages/backoffice/Loan/LoanDetails/LoanActions/ExtendModal.js

```diff
@@ -0,0 +1,79 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { Button } from 'semantic-ui-react';
+import { ConfirmModal } from '@components/ConfirmModal';
+import { withCancel } from '@api/utils';
+import { documentApi } from '@api/documents';
+
+export default class ExtendModal extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      overbooked: false,
+    };
+  }
+
+  componentDidMount() {
+    this.fetchCirculation();
+  }
+
+  componentWillUnmount() {
+    this.cancellableCirculation && this.cancellableCirculation.cancel();
+  }
+
+  fetchCirculation = async () => {
+    const { loanDetails, sendErrorNotification } = this.props;
+    const { document_pid } = loanDetails.metadata;
+    try {
+      this.cancellableCirculation = withCancel(documentApi.get(document_pid));
+      const response = await this.cancellableCirculation.promise;
+      const circulation = response.data.metadata.circulation;
+      this.setState({ overbooked: circulation.overbooked === true });
+    } catch (error) {
+      if (error !== 'UNMOUNTED') {
+        sendErrorNotification(
+          'We could not determine if the literature is overbooked.',
+          'Reload the page or contact the support if the problem persists.'
+        );
+      }
+    }
+  };
+
+  render() {
+    const { overbooked } = this.props;
+    return (
+      <>
+        {overbooked ? (
+          <ConfirmModal
+            header={`Extend Loan #${pid}`}
+            content={`Are you sure you want to extend the loan #${pid}?
+            The literature is overbooked.`}
+            buttonLabel="Extend"
+            action={loanAction}
+            isLoading={isLoading}
+            actionLabel="Extend"
+          />
+        ) : (
+          <Button
+            size="small"
+            fluid
+            primary
+            onClick={loanAction}
+            loading={isLoading}
+            disabled={isLoading}
+          >
+            Extend
+          </Button>
+        )}
+      </>
+    );
+  }
+}
+
+ExtendModal.propTypes = {
+  pid: PropTypes.string.isRequired,
+  isLoading: PropTypes.bool.isRequired,
+  loanAction: PropTypes.func.isRequired,
+  loanDetails: PropTypes.object.isRequired,
+  sendErrorNotification: PropTypes.func.isRequired,
```

Comment:
I'm not sure on passing as a prop the pid and the loanDetails. Maybe it's more efficient to pass only the loanDetails?
Also, I'm not 100% on passing the sendErrorNotification.

Reply:
you can map `sendErrorNotification` to the props of this component, exactly the way you did in the parent component, and you will not have to pass it down

the other two props I think are OK.